Contracts/tests/deposit and sell

W3stside opened the pull request:
### Deposit And Sell function testing

W3stside committed:
removed getter in favour of variable mapping call

W3stside committed:
added withdrawUnlockedTulip testing and passes

W3stside committed:
tweaking Travis script

W3stside committed:
fixed eventWatching issue in Tulips - changed toFixed(3) to 2 in assert as evm_mine arbitrary time add is causing failure

W3stside committed:
added in missing .solcover config file

W3stside committed:
added changing owner and changing minter tests

W3stside committed:
created depositAndSell test

Velenir commented on test/dutchExchange-DepositAndSell.js:
```diff
@@ -0,0 +1,68 @@
+
+    await Promise.all(testingAccs.map(acc => Promise.all([
+      dx.deposit(eth.address, ETHBalance / 2, { from: acc }),
+    ])))
```
Can do without the second `Promise.all` here.

W3stside replied:
yep, refactored this and didn't catch it. thx

Velenir commented on test/dutchExchange-DepositAndSell.js:
```diff
@@ -0,0 +1,68 @@
+    ])))
+  })
+
+  it('Adds Token Pair', async () => dx.addTokenPair(eth.address, gno.address, 0, 0, 2, 1, { from: accounts[1] }))
```
unnecessary `async`

W3stside replied:
woops. what I get for 330am coding. thx

Velenir commented on test/dutchExchange-DepositAndSell.js:
```diff
@@ -0,0 +1,68 @@
+  })))
+
+  it('DX balances cannot be more than amt deposited initially', () => Promise.all(testingAccs.map(async (acc) => {
+    assert(await dx.balances.call(eth.address, acc) <= ETHBalance / 2, 'Balances cannot be more than ETHBalance / 2')
```
Can use `assert.isAtMost` here, but whatever your preference.

Velenir commented on test/dutchExchange-Tulip.js:
```diff
@@ -662,16 +667,427 @@ const c2 = () => contract('DutchExchange --> Tulip Flow --> 1 Seller + 2 Buyers
+  before(async () => {
+    // get contracts
+    await setupContracts()
+    eventWatcher(dx, 'LogNumber', {})
```
Actually you don't need the last argument, it's `{}` by default.

Velenir commented on test/dutchExchange-Tulip.js:
```diff
@@ -662,16 +667,427 @@ const c2 = () => contract('DutchExchange --> Tulip Flow --> 1 Seller + 2 Buyers
+
+  it(
+    'BUYER1: Tries to lock and unlock Tulips --> Auction NOT cleared --> asserts 0 Tulips minted and in mapping', 
+    async () => unlockTulipTokens(buyer1),
```
unnecessary `async`, favorite keyword?

W3stside replied:
no, just tired and not thinking properly, also yes

- will fix these in #160

Velenir commented on test/dutchExchange-Tulip.js:
```diff
@@ -192,6 +194,11 @@ const c1 = () => contract('DutchExchange --> Tulip Flow --> 1 Seller + 1 Buyer |
 
+  it(
+    'BUYER1: Tries to lock and unlock Tulips --> Auction NOT cleared --> asserts 0 Tulips minted and in mapping', 
+    async () => unlockTulipTokens(buyer1),
```
unnecessary `async`

Velenir commented on test/dutchExchange-Tulip.js:
```diff
@@ -662,16 +667,427 @@ const c2 = () => contract('DutchExchange --> Tulip Flow --> 1 Seller + 2 Buyers
+  c4()
 } else {
-  return Promise.all([c1(), c2()])
+  return Promise.all([c1(), c2(), c3(), c4()])
```
Are you sure we need `Promise.all` here. Don't `contract()` just schedule tests and are run sequentially.

W3stside replied:
Haven't tested this, will check in #160

W3stside replied:
yeah `forEach` is fine

W3stside committed:
fixing errors per PR review -
1. removed unnecessary async fn keywords
2. removed unnecessary Promise.all blocks
